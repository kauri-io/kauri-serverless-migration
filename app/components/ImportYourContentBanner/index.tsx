import styled from 'styled-components'
import { Title2, BodyCard } from '../Typography'
import Button from '../../components/Button'

const Container = styled.section`
    display: flex;
    width: 100%;
    justify-content: center;
    align-items: center;
    padding: ${props => props.theme.space[3]}px;
    background: ${props => props.theme.colors.bgSecondary};
    @media (max-width: ${props => props.theme.breakpoints[0]}) {
        display: none;
    }
`

const Content = styled.section`
    display: flex;
    width: 500px;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    > :not(:last-child) {
        margin-bottom: ${props => props.theme.space[2]}px;
    }
`

const ImportBannerCopy = styled.div`
    display: flex;
    flex-direction: column;
    > :first-child {
        margin-bottom: ${props => props.theme.space[1]}px;
    }
`

const NewsletterBanner: React.FunctionComponent = _ => {
    return (
        <Container>
            <Content>
                <Title2 color="white">Import your content into Kauri</Title2>
                <ImportBannerCopy>
                    <BodyCard color="white" textAlign={'center'}>
                        Import a single article or your entire Medium
                        profile/publication in under 5 minutes with our
                        importer!
                    </BodyCard>
                    <BodyCard color="white" textAlign={'center'}>
                        Coming soon: Wordpress importer
                    </BodyCard>
                </ImportBannerCopy>
                <Button
                    color="secondary"
                    variant="outlined"
                    onClick={() => {
                        window.location.href = 'https://import.kauri.io'
                    }}
                >
                    Import Content
                </Button>
            </Content>
        </Container>
    )
}

export default NewsletterBanner
