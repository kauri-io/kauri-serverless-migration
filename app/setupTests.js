import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import theme from './lib/theme-config'
import { ThemeProvider } from 'styled-components'

Enzyme.configure({ adapter: new Adapter() })

const observeMock = {
    observe: () => null,
    disconnect: () => null, // maybe not needed
}

window.IntersectionObserver = () => observeMock

export const mountWithTheme = children =>
    mount(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const renderWithTheme = children =>
    render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const shallowWithTheme = children =>
    shallow(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
