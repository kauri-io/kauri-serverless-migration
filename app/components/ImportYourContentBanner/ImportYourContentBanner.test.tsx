import ImportYourContentBanner from './index'

import { mountWithTheme } from '../../setupTests'

describe('components/ImportYourContentBanner', () => {
    it('snapshot should match', () => {
        const wrapper = mountWithTheme(<ImportYourContentBanner />)
        expect(wrapper).toMatchSnapshot()
    })
})
