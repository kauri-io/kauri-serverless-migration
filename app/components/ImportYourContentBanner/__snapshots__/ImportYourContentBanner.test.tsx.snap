// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ImportYourContentBanner snapshot should match 1`] = `
<ThemeProvider
  theme={
    Object {
      "DEFAULT_CARD_WIDTH": 288,
      "MAIN_BODY_WIDTH": 1242,
      "bg": Object {
        "bgPrimary": "#1E2428",
        "bgSecondary": "#1E3D3B",
        "borderTextArea": "#979797",
        "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
        "disabledBackgroundColor": "#D6D6D6",
        "disabledTextColor": "#9B9B9B",
        "divider": "#EBEBEB",
        "errorRedColor": "#C03030",
        "hoverTextColor": "#108E72",
        "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
        "paddingTop": "2.5em",
        "primary": "#0BA986",
        "primaryColor": "#0BA986",
        "primaryDark": "#11856B",
        "primaryTextColor": "#1E2428",
        "secondary": "#11856B",
        "secondaryBlue": "rgb(81, 123, 162, 1)",
        "secondaryBlueDark": "rgb(81, 118, 142, 1)",
        "secondaryColor": "#1E3D3B",
        "secondaryTextColor": "#283035",
        "tertiaryBackgroundColor": "#F7F7F7",
        "textPrimary": "#1E2428",
        "white": "#FFFFFF",
      },
      "bgPrimary": "#1E2428",
      "bgSecondary": "#1E3D3B",
      "borderTextArea": "#979797",
      "breakpoints": Array [
        "500px",
        "52em",
        "1242px",
      ],
      "colors": Object {
        "bgPrimary": "#1E2428",
        "bgSecondary": "#1E3D3B",
        "borderTextArea": "#979797",
        "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
        "disabledBackgroundColor": "#D6D6D6",
        "disabledTextColor": "#9B9B9B",
        "divider": "#EBEBEB",
        "errorRedColor": "#C03030",
        "hoverTextColor": "#108E72",
        "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
        "paddingTop": "2.5em",
        "primary": "#0BA986",
        "primaryColor": "#0BA986",
        "primaryDark": "#11856B",
        "primaryTextColor": "#1E2428",
        "secondary": "#11856B",
        "secondaryBlue": "rgb(81, 123, 162, 1)",
        "secondaryBlueDark": "rgb(81, 118, 142, 1)",
        "secondaryColor": "#1E3D3B",
        "secondaryTextColor": "#283035",
        "tertiaryBackgroundColor": "#F7F7F7",
        "textPrimary": "#1E2428",
        "white": "#FFFFFF",
      },
      "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
      "disabledBackgroundColor": "#D6D6D6",
      "disabledTextColor": "#9B9B9B",
      "divider": "#EBEBEB",
      "errorRedColor": "#C03030",
      "fontSizes": Array [
        11,
        13,
        14,
        16,
        18,
        20,
        22,
        28,
        32,
      ],
      "fontWeight": Array [
        300,
        "normal",
        500,
        700,
      ],
      "hoverTextColor": "#108E72",
      "padContent": "
    max-width: 1242px;
    margin-left: auto;
    margin-right: auto;
    @media (min-width: 1242px) {
      min-width: 1242px;
    }
    @media (max-width: 1242px) {
      min-width: 90%;
    }
  ",
      "padding": "calc((100vw - 1242px) / 2)",
      "paddingTop": "2.5em",
      "primary": "#0BA986",
      "primaryColor": "#0BA986",
      "primaryDark": "#11856B",
      "primaryTextColor": "#1E2428",
      "secondary": "#11856B",
      "secondaryBlue": "rgb(81, 123, 162, 1)",
      "secondaryBlueDark": "rgb(81, 118, 142, 1)",
      "secondaryColor": "#1E3D3B",
      "secondaryTextColor": "#283035",
      "space": Array [
        0,
        10,
        15,
        30,
        60,
      ],
      "tertiaryBackgroundColor": "#F7F7F7",
      "textPrimary": "#1E2428",
      "white": "#FFFFFF",
    }
  }
>
  <NewsletterBanner>
    <ImportYourContentBanner__Container>
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "ImportYourContentBanner__Container-sc-83m2xu-0",
              "isStatic": false,
              "lastClassName": "fECYWm",
              "rules": Array [
                "display:flex;width:100%;justify-content:center;align-items:center;padding:",
                [Function],
                "px;background:",
                [Function],
                ";@media (max-width:",
                [Function],
                "){display:none;}",
              ],
            },
            "displayName": "ImportYourContentBanner__Container",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "ImportYourContentBanner__Container-sc-83m2xu-0",
            "target": "section",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <section
          className="ImportYourContentBanner__Container-sc-83m2xu-0 fECYWm"
        >
          <ImportYourContentBanner__Content>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "ImportYourContentBanner__Content-sc-83m2xu-1",
                    "isStatic": false,
                    "lastClassName": "nrBSt",
                    "rules": Array [
                      "display:flex;width:500px;flex-direction:column;justify-content:center;align-items:center;>:not(:last-child){margin-bottom:",
                      [Function],
                      "px;}",
                    ],
                  },
                  "displayName": "ImportYourContentBanner__Content",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "ImportYourContentBanner__Content-sc-83m2xu-1",
                  "target": "section",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <section
                className="ImportYourContentBanner__Content-sc-83m2xu-1 nrBSt"
              >
                <Typography__styledComponent
                  color="white"
                >
                  <StyledComponent
                    color="white"
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "Typography__styledComponent-sc-36xkf6-5",
                          "isStatic": false,
                          "lastClassName": "jNVtUZ",
                          "rules": Array [
                            "color:",
                            [Function],
                            ";margin:0px;font-weight:",
                            "500",
                            ";",
                            ";font-size:",
                            "20",
                            "px;text-transform:",
                            [Function],
                            ";text-align:",
                            [Function],
                            ";:hover{color:",
                            [Function],
                            ";cursor:",
                            [Function],
                            ";}",
                          ],
                        },
                        "displayName": "Typography__styledComponent",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "Typography__styledComponent-sc-36xkf6-5",
                        "target": "span",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <span
                      className="Typography__styledComponent-sc-36xkf6-5 jNVtUZ"
                      color="white"
                    >
                      Import your content into Kauri
                    </span>
                  </StyledComponent>
                </Typography__styledComponent>
                <ImportYourContentBanner__ImportBannerCopy>
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "ImportYourContentBanner__ImportBannerCopy-sc-83m2xu-2",
                          "isStatic": false,
                          "lastClassName": "jCDnUQ",
                          "rules": Array [
                            "display:flex;flex-direction:column;>:first-child{margin-bottom:",
                            [Function],
                            "px;}",
                          ],
                        },
                        "displayName": "ImportYourContentBanner__ImportBannerCopy",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "ImportYourContentBanner__ImportBannerCopy-sc-83m2xu-2",
                        "target": "div",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <div
                      className="ImportYourContentBanner__ImportBannerCopy-sc-83m2xu-2 jCDnUQ"
                    >
                      <Typography__component
                        color="white"
                        textAlign="center"
                      >
                        <StyledComponent
                          color="white"
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "Typography__component-sc-36xkf6-4",
                                "isStatic": false,
                                "lastClassName": "chprKh",
                                "rules": Array [
                                  "font-size:14px;font-weight:normal;letter-spacing:-0.1px;margin-bottom:",
                                  [Function],
                                  "px;line-height:18px;color:",
                                  [Function],
                                  ";",
                                  ";",
                                  [Function],
                                  ";",
                                ],
                              },
                              "displayName": "Typography__component",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "Typography__component-sc-36xkf6-4",
                              "target": "span",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                          textAlign="center"
                        >
                          <span
                            className="Typography__component-sc-36xkf6-4 chprKh"
                            color="white"
                          >
                            Import a single article or your entire Medium profile/publication in under 5 minutes with our importer!
                          </span>
                        </StyledComponent>
                      </Typography__component>
                      <Typography__component
                        color="white"
                        textAlign="center"
                      >
                        <StyledComponent
                          color="white"
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "Typography__component-sc-36xkf6-4",
                                "isStatic": false,
                                "lastClassName": "chprKh",
                                "rules": Array [
                                  "font-size:14px;font-weight:normal;letter-spacing:-0.1px;margin-bottom:",
                                  [Function],
                                  "px;line-height:18px;color:",
                                  [Function],
                                  ";",
                                  ";",
                                  [Function],
                                  ";",
                                ],
                              },
                              "displayName": "Typography__component",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "Typography__component-sc-36xkf6-4",
                              "target": "span",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                          textAlign="center"
                        >
                          <span
                            className="Typography__component-sc-36xkf6-4 chprKh"
                            color="white"
                          >
                            Coming soon: Wordpress importer
                          </span>
                        </StyledComponent>
                      </Typography__component>
                    </div>
                  </StyledComponent>
                </ImportYourContentBanner__ImportBannerCopy>
                <SecondaryButtonComponent
                  onClick={[Function]}
                >
                  <SecondaryButton
                    bg="transparent"
                    border="white"
                    borderHover="primary"
                    color="white"
                    fontSize={0}
                    fontWeight={700}
                    mr={2}
                    onClick={[Function]}
                    type="button"
                  >
                    <StyledComponent
                      bg="transparent"
                      border="white"
                      borderHover="primary"
                      color="white"
                      fontSize={0}
                      fontWeight={700}
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "SecondaryButton-sy4hb2-0",
                            "isStatic": false,
                            "lastClassName": "faBvdL",
                            "rules": Array [
                              "display:flex;align-items:center;justify-content:center;min-width:136px;max-width:150px;",
                              [Function],
                              ";height:40px;border-radius:4px;border:none;cursor:pointer;> svg,img{height:16px;",
                              [Function],
                              ";}text-transform:uppercase;opacity:",
                              [Function],
                              ";",
                              [Function],
                              ";",
                              [Function],
                              ";",
                              [Function],
                              ";",
                              [Function],
                              ";:hover{cursor:",
                              [Function],
                              ";}",
                              ";box-shadow:0px 0px 0px 1px ",
                              [Function],
                              ";:hover{box-shadow:0px 0px 0px 2px ",
                              [Function],
                              ";}",
                            ],
                          },
                          "displayName": "SecondaryButton",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "SecondaryButton-sy4hb2-0",
                          "target": "button",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      mr={2}
                      onClick={[Function]}
                      type="button"
                    >
                      <button
                        className="SecondaryButton-sy4hb2-0 faBvdL"
                        color="white"
                        fontSize={0}
                        fontWeight={700}
                        onClick={[Function]}
                        type="button"
                      >
                        Import Content
                      </button>
                    </StyledComponent>
                  </SecondaryButton>
                </SecondaryButtonComponent>
              </section>
            </StyledComponent>
          </ImportYourContentBanner__Content>
        </section>
      </StyledComponent>
    </ImportYourContentBanner__Container>
  </NewsletterBanner>
</ThemeProvider>
`;
